File: javascript/nth-prime/nth-prime.js
export class Prime {
  nth(n) {
    if (n < 1) {
      throw new Error('Prime is not possible');
    }
    let i = 1;
    while (n > 0) {
      i++;
      if (this.isPrime(i)) {
        n--;
      }
    }
    return i;
  }

  isPrime(i) {
    for (let factor = 2; factor <= Math.floor(Math.sqrt(i)); factor++) {
      if (i % factor === 0) {
        return false;
      }
    }
    return true;
  }
}

File: ruby/nth-prime/nth_prime.rb
module Prime
  def self.nth(n)
    raise ArgumentError, "n must be positive" unless n.positive?
    return 2 if n == 1

    n -= 1
    x = 3
    loop do
      n -= 1 if prime? x
      return x if n.zero?
      x += 2
    end
  end

  def self.prime?(x)
    (2..Math.sqrt(x)).none? { |d| x % d == 0 }
  end
end
